src/rust-crypto/backup.ts

```diff
@@ -124,6 +145,43 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         }
         return this.keyBackupCheckInProgress;
     }
+
+    /**
+     * Handles a backup secret received event and store it if it matches the current backup version.
+     *
+     * @param secret - The secret as received from a `m.secert.send` event for secret `m.megolm_backup.v1`.
+     * @returns true if the secret is valid and has been stored, false otherwise.
+     */
+    public async handleBackupSecretReceived(secret: string): Promise<boolean> {
+        // Currently we only receive the decryption key without any key backup version, it is important to
+        // check that the secret is valid for the current version before storing it.
+        // We force a check to ensure to have the latest version. We also want to check that the backup is trusted
+        // as we don't want to store the secret if the backup is not trusted, and eventually import megolm keys later from an untrusted backup.
+        const backupCheck = await this.checkKeyBackupAndEnable(true);
```

Comment:
Not sure to get it, are you suggesting to skip the checks?. As explained in the comment we need to check for the current backup and we want it to be trusted. So yes if it's trusted it will enable it, can't hurt.

>  clients still using an old backup version

This is not possible, there is only one backup you can upload to.

Reply:
yeah, fair enough.